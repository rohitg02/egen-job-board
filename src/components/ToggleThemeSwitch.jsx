import React, { useContext }  from 'react';

import { Switch } from '@material-ui/core';
import WbSunnyRoundedIcon from '@material-ui/icons/WbSunnyRounded';
import Brightness2RoundedIcon from '@material-ui/icons/Brightness2Rounded';
import ThemeContext from '../contexts/ThemeContext';

const ToggleThemeSwitch = () => {
  const sampleContext= useContext(ThemeContext);
  return (
    <React.Fragment>
      <WbSunnyRoundedIcon/>
        <Switch
           id="themeSwitch"
           checked={sampleContext.theme === 'light' ? false : true }
           onChange={sampleContext.toggleTheme}
           size="small"
           name="toggleTheme"
           color="white"
         />
       <Brightness2RoundedIcon/>
    </React.Fragment>
  );
}
//chnage color white prop
export default ToggleThemeSwitch;
