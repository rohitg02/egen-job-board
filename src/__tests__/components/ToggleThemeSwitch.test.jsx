import React from 'react';
import { shallow } from '../../enzyme';

import ToggleThemeSwitch from '../../components/ToggleThemeSwitch';
import ThemeContext from '../../contexts/ThemeContext';
import light from '../../themes/light';

describe('ToggleThemeSwitch component', () => {

    it('should render ToggleThemeSwitch component', () => {
      const wrapper = shallow(<ToggleThemeSwitch />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should render ToggleThemeSwitch component with context and set the check to false', () => {
      const contextValue = {
        ...light,
        toggleTheme: jest.fn()
      }
      let useContextMock = React.useContext = jest.fn();
      useContextMock.mockReturnValue(contextValue);
      const wrapper = shallow(<ToggleThemeSwitch />, useContextMock);
      expect(wrapper.find('#themeSwitch').props().checked).toEqual(false);
    });

});
