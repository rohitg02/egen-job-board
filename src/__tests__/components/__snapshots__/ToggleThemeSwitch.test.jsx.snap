// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedSearchField should render ThemedSearchField component and match the snapshot 1`] = `ShallowWrapper {}`;

exports[`ThemedSearchField should render ToggleThemeSwitch component 1`] = `ShallowWrapper {}`;

exports[`ToggleThemeSwitch component should render ToggleThemeSwitch component 1`] = `ShallowWrapper {}`;
